code/run_capsule.py

```diff
@@ -1,417 +1,415 @@
+    plt.hist(data1[:, -1], bins=100, range=(255, 270), color="green", alpha=0.7)
+    plt.xlim(255, 270)
+    GreenChFloorAve = np.mean(data1[:, -1])
+    plt.title(f"GreenCh FloorAve: {GreenChFloorAve:.2f}")
+    plt.xlabel("CMOS pixel val")
+    plt.ylabel("counts")
+
+    # IsoCh Floor
+    plt.subplot(1, 3, 2)
+    plt.hist(data2[:, -1], bins=100, range=(255, 270), color="purple", alpha=0.7)
+    plt.xlim(255, 270)
+    IsoChFloorAve = np.mean(data2[:, -1])
+    plt.title(f"IsoCh FloorAve: {IsoChFloorAve:.2f}")
+    plt.xlabel("CMOS pixel val")
+    plt.ylabel("counts")
+
+    # RedCh Floor
+    plt.subplot(1, 3, 3)
+    plt.hist(data3[:, -1], bins=100, range=(255, 270), color="red", alpha=0.7)
+    plt.xlim(255, 270)
+    RedChFloorAve = np.mean(data3[:, -1])
+    plt.title(f"RedCh FloorAve: {RedChFloorAve:.2f}")
+    plt.xlabel("CMOS pixel val")
+    plt.ylabel("counts")
+
+    plt.subplots_adjust(wspace=0.8)
+
+    # Save and show the plot
+    plt.savefig(f"{results_folder}/CMOS_Floor.png", dpi=300, bbox_inches="tight")
+    plt.savefig(f"{results_folder}/CMOS_Floor.pdf")
+    plt.show()
+
+
+def plot_sync_pulse_diff(rising_time, results_folder):
+    """
+    Plot a histogram of the differences in rising times and save the plot.
+
+    Parameters:
+        rising_time (array-like): An array of rising time values.
+        save_path (str): The path to save the generated plot.
+    """
+    # Compute differences
+    diffs = np.diff(rising_time)
+
+    # Create the plot
+    plt.figure()
+    plt.hist(diffs, bins=100, range=(0, 0.2))
+    plt.title("sync pulse diff")
+    plt.ylabel("counts")
+    plt.xlabel("ms")
+
+    # Save and show the plot
+    plt.savefig(f"{results_folder}/SyncPulseDiff.png")
+    plt.savefig(f"{results_folder}/SyncPulseDiff.pdf")
+    plt.show()
+
+
+def main():
+    # Paths and setup
+    fiber_base_path = Path("/data/fiber_raw_data")
+    fiber_raw_path = fiber_base_path / "fib"
+    results_folder = Path("../results")
+    results_folder.mkdir(parents=True, exist_ok=True)
+
+    # Load JSON files
+    subject_data = load_json_file(fiber_base_path / "subject.json")
+    subject_id = subject_data.get("subject_id")
+    if not subject_id:
+        logging.info("Error: Subject ID is missing from subject.json.")
+        sys.exit(1)
+
+    date_data = load_json_file(fiber_base_path / "data_description.json")
+    date = date_data.get("creation_time")
+    if not date:
+        session_data = load_json_file(fiber_base_path / "session.json")
+        date = session_data.get("session_start_time")
+
+    asset_name = f"behavior_{subject_id}_{date}"
+    setup_logging("aind-fip-qc-raw", mouse_id=subject_id, session_name=asset_name)
+
+    try:
+        channel_file_paths = [
+            next(fiber_raw_path.glob(fiber_channel))
+            for fiber_channel in ["FIP_DataG*", "FIP_DataIso_*", "FIP_DataR_*"]
+        ]
+    except StopIteration:
+        logging.info("Error: FIP data files are missing.")
```

Comment:
I would suggest to fire a logging.error instead of .info there. Instead of sys.exit() maybe let the error escalate up.